sunpy/map/mapbase.py

```diff
@@ -1520,34 +1521,28 @@ def submap(self, bottom_left, *, top_right=None, width: (u.deg, u.pix) = None, h
             [268.24377, 254.83157, 268.24377, 321.89252],
             [249.99167, 265.14267, 274.61206, 240.5223 ]], dtype=float32)
         """
-        if not isinstance(bottom_left, u.Quantity):
-            bottom_left, top_right = get_rectangle_coordinates(bottom_left,
-                                                               top_right=top_right,
-                                                               width=width,
-                                                               height=height)
+        # Handle the case where bottom_left and top_right are really in bottom_left
+        if isinstance(bottom_left, SkyCoord) and bottom_left.shape == (2,):
+            top_right = bottom_left[1]
+            bottom_left = bottom_left[0]
```

Comment:
This is done in the `get_rectangle_helper` right? Why are we repeating it here? Can't we pass `top_right` through as none to the generic function?